### DISCUSSION TITLE
More pretty and readable markdown for chat window.

### DISCUSSION THREAD

**Discussion by PLAZMAMA** [olimorris/codecompanion.nvim#358] [Show and tell]
**Limitation:** Rendering/prettifying only happens after the output is completely generated.

https://github.com/user-attachments/assets/6878c002-70e3-4c81-815e-1abc390fdea3

**Markdown Extension:** [render-markdown.nvim](https://github.com/MeanderingProgrammer/render-markdown.nvim)

**Comment by olimorris** [comment]
Did you have to do any specific configuration for CodeCompanion?

**Reply by PLAZMAMA** [reply]
I had to add "codecompanion" as a file type to the extension. Here is my config:

```lua
return {
    -- Make sure to set this up properly if you have lazy=true
    'MeanderingProgrammer/render-markdown.nvim',
    lazy = true,
    version = 'v7.x.x',
    ft = { 'markdown', 'codecompanion' },
    config = function()
        require('render-markdown').setup({
            render_modes = true,
        })
    end,
}
```

**Reply by olimorris** [reply]
Have you noticed any performance issues? I believe that plugin applies a lot of extmarks and autocmds.

**Reply by PLAZMAMA** [reply]
not at all, but then again I'm using an M1 Pro so its pretty fast for most things.

**Reply by bassamsdata** [reply]
regarding performance, from my experience, I'd say it's a very performant plugin, even on an old laptop, and it doesn't cause any issues. Actually, I hold the plugin in high regard for its exceptional performance.

**Comment by bassamsdata** [comment]
I don't know why, but I thought the CodeCompanion readme already recommend using the plugin, and I assumed people were using render-markdown or something similar by default. 
Turns out, I was mistaken! I've been using it with this plugin from the start, and yes, it definitely enhances the look a lot beside it's really fast and performant.

**Reply by PLAZMAMA** [reply]
I think that it might be a good idea to add it to the documentation, even as just a recommendation.

**Comment by olimorris** [comment]
I've "added" support for this plugin in the latest release (`v9.1.0`). You can turn off any rendering of headers by CodeCompanion with:

```lua
require("codecompanion").setup({
  display = {
    chat = {
      render_headers = false,
    }
  }
})
```

Also if you happen to use [OneDarkPro.nvim](https://github.com/olimorris/onedarkpro.nvim), I've added support for `rendering-markdown.nvim` for that.

**Reply by PLAZMAMA** [reply]
Thank you for this. However, I am ignorant of what turning off the headers rendering does. I guess it's time for me to find out.
